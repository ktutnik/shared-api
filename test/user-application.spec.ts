import supertest from "supertest"
import createApp from "../src/app"
import { closeConnection, createApplication, createUser, ignoreProps } from "./helper"

afterEach(async () => {
    await closeConnection()
})

describe("User Application", () => {
    it("Should able to list all user application", async () => {
        const plum = await createApp({ mode: "production" })
        const { app, owner } = await createApplication(plum)
        const { body } = await supertest(plum.callback())
            .post("/api/backend/applications")
            .set("Authorization", `Bearer ${owner.token}`)
            .send({ name: "Ipsum Lorem" })
            .expect(200)
        await supertest(plum.callback())
            .post(`/api/backend/applications/${body.id}/activate`)
            .set("Authorization", `Bearer ${owner.token}`)
            .send({ payment: 20 })
            .expect(200)
        const { body:result } = await supertest(plum.callback())
            .get(`/api/backend/users/${app.id}/applications`)
            .set("Authorization", `Bearer ${owner.token}`)
            .expect(200)
        expect(result[0]).toMatchSnapshot({...ignoreProps, apiKey: expect.any(String)})
        expect(result[1]).toMatchSnapshot({...ignoreProps, apiKey: expect.any(String)})
    })
})