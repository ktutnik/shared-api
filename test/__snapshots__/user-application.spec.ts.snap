// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Application Should able to list all user application 1`] = `
Object {
  "active": true,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Lorem Ipsum",
  "type": "Basic",
  "updatedAt": Any<String>,
}
`;

exports[`User Application Should able to list all user application 2`] = `
Object {
  "active": true,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Ipsum Lorem",
  "type": "Premium",
  "updatedAt": Any<String>,
}
`;
